x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_esql.tsx

```diff
@@ -98,7 +98,8 @@ export const IndexDataVisualizerESQL: FC<IndexDataVisualizerESQLProps> = (dataVi
   const indexPattern = useMemo(() => {
     let indexPatternFromQuery = '';
     if (isESQLQuery(query)) {
-      indexPatternFromQuery = getIndexPatternFromESQLQuery(query.esql);
+      // deprecated, use the async new getIndexPatternFromESQLQuery instead
+      indexPatternFromQuery = getIndexPatternFromESQLQueryDeprecated(query.esql);
```

Comment:
ℹ️ same here, we need to use the `getIndexPatternFromESQLQuery ` instead which is asynchronous. I will wait for the ML team to handle it.